packages/block-editor/src/store/reducer.js

```diff
@@ -1228,17 +1205,44 @@ export const blockListSettings = ( state = {}, action ) => {
 	return state;
 };
 
-export default withPostMetaUpdateCacheReset(
-	combineReducers( {
-		blocks,
-		isTyping,
-		isCaretWithinFormattedText,
-		blockSelection,
-		blocksMode,
-		blockListSettings,
-		insertionPoint,
-		template,
-		settings,
-		preferences,
-	} )
-);
+/**
+ * Reducer return an updated state representing the most recent block attribute
+ * update. The state is structured as an object where the keys represent the
+ * client IDs of blocks, the values a subset of attributes from the most recent
+ * block update. The state is always reset to null if the last action is
+ * anything other than an attributes update.
+ *
+ * @param {Object<string,Object>} state  Current state.
+ * @param {Object}                action Action object.
+ *
+ * @return {[string,Object]} Updated state.
+ */
+export function lastBlockAttributesChange( state, action ) {
+	switch ( action.type ) {
+		case 'UPDATE_BLOCK':
+			if ( ! action.updates.attributes ) {
+				break;
+			}
+
+			return { [ action.clientId ]: action.updates.attributes };
+
+		case 'UPDATE_BLOCK_ATTRIBUTES':
+			return { [ action.clientId ]: action.attributes };
+	}
```

Comment:
What happens for other actions that change multiple blocks, like RESET_BLOCKS... (Thinking about a third-party plugin triggering that action and not the block editor provider), should we try to list the changes or do we just trigger a full sync? What happens now?

Reply:
> What happens for other actions that change multiple blocks, like RESET_BLOCKS... (Thinking about a third-party plugin triggering that action and not the block editor provider), should we try to list the changes or do we just trigger a full sync? What happens now?

It wouldn't try to sync meta attributes from the updates, which is the same as how it behaves today in master. I think it might be a bit difficult to infer from the reset whether those meta values should be considered canonical, or whether the source should be applied again by the editor. The latter is what I expect happens in the current state of this branch.